sympy/core/function.py

```diff
@@ -2824,19 +2824,105 @@ def _eval_expand_hint(self, **hints):
 
 # This is a special application of two hints
 
-def _mexpand(expr, recursive=False):
-    # expand multinomials and then expand products; this may not always
-    # be sufficient to give a fully expanded expression (see
-    # test_issue_8247_8354 in test_arit)
-    if expr is None:
-        return
-    was = None
-    while was != expr:
-        was, expr = expr, expand_mul(expand_multinomial(expr))
-        if not recursive:
-            break
-    return expr
+def mexpand(expr, recursive=False, *, numer=None, denom=None, frac=None):
+    """expand multinomials and then expand products in ``expr``.
+
+    If recursive is False, this may not always be sufficient to
+    give a fully expanded expression.
+
+    To target expansion only on the numerator or denominator, set
+    ``numer`` or ``denom`` to True, respectively. To keep numerator
+    and denominator over each other and expand each, make `frac`` True.
+
+    Examples
+    ========
+
+    >>> from sympy.core.function import mexpand
+    >>> from sympy.abc import x
+    >>> eq = (x - 1)**2/(1 - x)**2
+    >>> mexpand(eq, frac=True)
+    1
+    >>> mexpand(eq, numer=True)
+    (x**2 - 2*x + 1)/(1 - x)**2
+    >>> mexpand(eq, denom=True)
+    (x - 1)**2/(x**2 - 2*x + 1)
+    >>> mexpand(eq)
+    x**2/(x**2 - 2*x + 1) - 2*x/(x**2 - 2*x + 1) + 1/(x**2 - 2*x + 1)
+
+    >>> eq -= 1
+    >>> mexpand(eq, frac=True)
+    0
+    >>> mexpand(eq, numer=True)
+    0
+    >>> mexpand(eq, denom=True)
+    -1 + (x - 1)**2/(1 - x)**2
+    >>> mexpand(eq)
+    0
+
+    See Also
+    ========
+    mexpand_cse
```

Comment:
We also want the see also for `expand` I think which in turn should reference all the `expand_*` functions. It probably makes sense to add a see also to `expand` linking back here as well.